src/complex/Pipeline/PipelineFilter.cpp

```diff
@@ -96,11 +99,13 @@ bool PipelineFilter::preflight(DataStructure& data)
     for(auto&& warning : actionResult.warnings())
     {
       m_Warnings.push_back(std::move(warning));
+      setHasWarnings(true);
```

Comment:
I figured it was redundant to make a call to check if it's empty since we push back to it right before it but sure I can do that.

Reply:
I think there will be some redundancy either way since it only needs to be set once. Also looking at it again it should probably come after the loop and use `m_Warnings` instead. Otherwise the original change that I suggested will reset the warning state even if they were warnings from preflight but not the actions. I missed that on my first look.

Reply:
Ah yeah you're right.  I guess that would be the more correct way to set it.  Thanks